test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/LabelTagHelperTest.cs

```diff
@@ -234,6 +230,109 @@ public class LabelTagHelperTest
             Assert.Equal(expectedTagName, output.TagName);
         }
 
+        // Display name, original child content, HTML field prefix, expected child content, and expected ID.
+        // Uses TagHelperOutputContent.OriginalContent to pass HtmlFieldPrefix values.
+        public static TheoryData<string, string, string, string, string> DisplayNameDataSet
+        {
+            get
+            {
+                return new TheoryData<string, string, string, string, string>
+                {
+                    {
+                        null, string.Empty, string.Empty, $"HtmlEncode[[{nameof(NestedModel.Text)}]]", nameof(NestedModel.Text)
+                    },
+                    {
+                        string.Empty, string.Empty, string.Empty, string.Empty, nameof(NestedModel.Text)
+                    },
+                    {
+                        "a label", string.Empty, string.Empty, "HtmlEncode[[a label]]", nameof(NestedModel.Text)
+                    },
+                    {
+                        null, "original label", string.Empty, "original label", nameof(NestedModel.Text)
+                    },
+                    {
+                        string.Empty, "original label", string.Empty, "original label", nameof(NestedModel.Text)
+                    },
+                    {
+                        "a label", "original label", string.Empty, "original label", nameof(NestedModel.Text)
+                    },
+                    {
+                        null, string.Empty, "prefix", $"HtmlEncode[[{nameof(NestedModel.Text)}]]", $"prefix_{nameof(NestedModel.Text)}"
+                    },
+                    {
+                        string.Empty, string.Empty, "prefix", string.Empty, $"prefix_{nameof(NestedModel.Text)}"
+                    },
+                    {
+                        "a label", string.Empty, "prefix", "HtmlEncode[[a label]]", $"prefix_{nameof(NestedModel.Text)}"
+                    },
+                };
+            }
+        }
+
+        // Prior to aspnet/Mvc#6638 fix, helpers generated nothing in this test when displayName was empty.
+        [Theory]
+        [MemberData(nameof(DisplayNameDataSet))]
```

Comment:
nit: Could totally just use InlineData

Reply:
Nah. I want to use `string.Empty` and interpolated strings.